tidy3d/constants.py

```diff
@@ -13,20 +13,22 @@
 
 import numpy as np
 
+# floating point types and precisions
+TIDY3D_DOUBLE_PRECISION = False
+fp_type = np.float64 if TIDY3D_DOUBLE_PRECISION else np.float32
+cfp_type = np.complex128 if TIDY3D_DOUBLE_PRECISION else np.complex64
+fp_eps = np.finfo(fp_type).eps
+
 # fundamental constants
-EPSILON_0 = np.float32(8.85418782e-18)
-MU_0 = np.float32(1.25663706e-12)
+EPSILON_0 = fp_type(8.85418782e-18)
```

Comment:
I think we can just remove the conversion here. Some of the explicit conversions to float came from the very early days in the solver, when I was kinda trying to make everything relevant to the solver single precision.